api/routes/generate.js

```diff
@@ -10,35 +10,30 @@ var thunkify = require('thunkify');
 // CRX packaging module, instantiated with the `new` keyword
 var Extension = require('crx');
 
+var knownHosts = {
+    facebookMessanger: 'messenger.com',
+    googleKeep: 'keep.google.com',
+    whatsappWeb: 'web.whatsapp.com'
+};
+
 // POST /generate
 module.exports = function *() {
     // Get target URL from input
     var url = this.request.body.url;
-    
+
     // Build .crx config for the provided URL
     var crxConfig = yield exports.buildCrxConfig(url);
-    
+
     // Generate the .crx file based on the config
     var crxBuffer = yield exports.generateCrx(crxConfig);
-    
+
     // Send it to the browser (save to disk)
     yield exports.sendCrx(this, crxConfig, crxBuffer);
-}
+};
 
 exports.buildCrxConfig = function *(targetUrl) {
-    // Bad input?
-    if (! targetUrl) {
-        throw new Error('Please provide a URL to continue.');
-    }
-    
-    // Parse URL (to retrieve hostname and verify its validity)
-    var parsedUrl = url.parse(targetUrl);
-    
-    // Parse failed?
-    if (! parsedUrl || ! parsedUrl.protocol || parsedUrl.protocol.indexOf('http') == -1) {
-        throw new Error('Please provide a valid URL for your extension. (It must start with http(s)://)');
-    }
-        
+    var parsedUrl = exports.validateUrl(targetUrl);
```

Comment:
true.
Renamed to parseUrl. 
And as any method should do it validates its input parameter and throws in case of invalid input.